graphql_compiler/schema_transformation/split_query.py

```
@@ -0,0 +1,642 @@
+    if child_type is None:
+        raise SchemaStructureError(
+            u'The provided merged schema descriptor may be invalid, as the type '
+            u'corresponding to the field "{}" under type "{}" cannot be '
+            u'found.'.format(ast.name.value, type_info.get_parent_type())
+        )
+    child_type_name = strip_non_null_and_list_from_type(child_type).name
+
+    child_selection_set = ast.selection_set
+    # Adjust for type coercion
+    if (
+        child_selection_set is not None and
+        len(child_selection_set.selections) == 1 and
+        isinstance(child_selection_set.selections[0], ast_types.InlineFragment)
+    ):
+        type_coercion_inline_fragment = child_selection_set.selections[0]
+        child_type_name = type_coercion_inline_fragment.type_condition.name.value
+        child_selection_set = type_coercion_inline_fragment.selection_set
+
+    return child_type_name, child_selection_set.selections
+
+
+def _get_edge_to_stitch_fields(merged_schema_descriptor):
+    """Get a map from type/field of each cross schema edge, to the fields that the edge stitches.
+
+    This is necessary only because graphql currently doesn't process schema directives correctly.
+    Once schema directives are correctly added to GraphQLSchema objects, this part may be
+    removed as directives on a schema field can be directly accessed.
+
+    Args:
+        merged_schema_descriptor: MergedSchemaDescriptor namedtuple, containing a schema ast
+                                  and a map from names of types to their schema ids
+
+    Returns:
+        Dict[Tuple(str, str), Tuple(str, str)], mapping (type name, edge field name) to
+        (source field name, sink field name) used in the @stitch directive, for each cross
+        schema edge
+    """
+    edge_to_stitch_fields = {}
+    for type_definition in merged_schema_descriptor.schema_ast.definitions:
+        if isinstance(type_definition, (
```

Should it be possible for `type_definition` to not be one of these two types? If not, what should happen if you get one that's not? If it shouldn't be possible, maybe raise an error instead of this conditional.

---

It's possible -- for instance, union type, scalars, directive definitions. We just ignore those, as they don't contain vertex fields (or any fields at all).